docs: Document Email content preview on the Email details page (#813)
mautic/user-documentation: docs/channels/emails.rst

The Email overview allows at-a-glance information regarding the success or failure of a particular Email. You can quickly see relevant information in regards to opens, bounces, successful click-throughs and other important statistics.

> .. _email content preview:

Reviewer: ```suggestion
.. _Email content preview:
```

Email content preview
======================

.. vale off

> The Email details page shows a rendered preview of the Email content in the right column, so you can see how an Email looks without opening the Builder or a separate tab. This is handy when you're comparing several Emails to find the one you want.

Author: [PR #15692](https://github.com/mautic/mautic/pull/15692) added a rendered preview `iframe` (`content_preview_frame`) to the right section of `app/bundles/EmailBundle/Resources/views/Email/details.html.twig`, replacing the old plain-text subject/preview header. I documented that the Email details page now shows a rendered content preview in the right column.

Source: https://github.com/mautic/mautic/pull/15692

> The 'Preview URL' panel below the preview controls what the preview shows. Select an A/B variant or translation from the 'Show preview for A/B variant' list to preview that version's content, or start typing in the 'Show preview for contact' field to render the Email with a specific Contact's tokens resolved.

Reviewer: ```suggestion
The **Preview URL** panel below the preview controls what the preview shows. Select an A/B variant or translation from the **Show preview for A/B variant** list to preview that version's content, or start typing in the **Show preview for contact** field to render the Email with a specific Contact's tokens resolved.
```

> The **Preview URL** panel below the preview controls what the preview shows. Select an A/B variant or translation from the **Show preview for A/B variant** list to preview that version's content, or start typing in the **Show preview for contact** field to render the Email with a specific Contact's tokens resolved.

Reviewer: I don't see the source code PR has changes about A/B variant. I have the same question for line 85. Where's this info coming from?

.. image:: images/emails/email_details_preview.png

>   :width: 400

Reviewer: Change the width to 600.

> Mautic reloads the preview automatically when you switch between A/B test variants or translations, so it always reflects the version you've selected.

Author: In [PR #15692](https://github.com/mautic/mautic/pull/15692), `app/bundles/CoreBundle/Assets/js/14.content.preview.js` updates the `content_preview_frame` `src` and re-runs `resizePreviewFrame()` whenever the preview URL changes, so the preview reloads automatically when switching A/B variants/translations or selecting a Contact. I documented this auto-reload behavior.

Source: https://github.com/mautic/mautic/pull/15692

>   :alt: Rendered Email content preview in the right column of the Email details page.

Reviewer: Please give align center.

  :width: 400

>   :alt: Screenshot showing the rendered Email content preview in the right column of the Email details page.

Reviewer: ```suggestion
  :alt: Rendered Email content preview in the right column of the Email details page.
```

The Email overview allows at-a-glance information regarding the success or failure of a particular Email. You can quickly see relevant information in regards to opens, bounces, successful click-throughs and other important statistics.

.. _Email content preview:

Email content preview
======================

.. vale off

The Email details page shows a rendered preview of the Email content in the right column, so you can see how an Email looks without opening the Builder or a separate tab. This is handy when you're comparing several Emails to find the one you want.

The preview reflects your selection in the **Preview URL** panel below it. Choose an A/B variant from **Show preview for A/B variant**, a translation from **Show preview for translation**, or enter a Contact in **Show preview for contact** to resolve that Contact's tokens. Mautic reloads the preview automatically to match the version you've selected.

.. image:: images/emails/email_details_preview.png
  :width: 600
  :align: center

>   :alt: Rendered Email content preview in the right column of the Email details page.

Reviewer: Add `|` to give some space with the following section and adjust the indentation.

```suggestion
.. image:: images/emails/email_details_preview.png
   :width: 600
   :align: center
   :alt: Rendered Email content preview in the right column of the Email details page.
  
|
```

The Email overview allows at-a-glance information regarding the success or failure of a particular Email. You can quickly see relevant information in regards to opens, bounces, successful click-throughs and other important statistics.

.. _Email content preview:

Email content preview
======================

.. vale off

The Email details page shows a rendered preview of the Email content in the right column, so you can see how an Email looks without opening the Builder or a separate tab. This is handy when you're comparing several Emails to find the one you want.

> The preview reflects your selection in the **Preview URL** panel below it. Choose an A/B variant from **Show preview for A/B variant**, a translation from **Show preview for translation**, or enter a Contact in **Show preview for contact** to resolve that Contact's tokens. Mautic reloads the preview automatically to match the version you've selected.

Reviewer: What does "to resolve that Contact's tokens" means here? Does general user understand 'contact's token' while we haven't mentioned it anywhere?

> The preview reflects your selection in the **Preview URL** panel below it. Choose an A/B variant from **Show preview for A/B variant**, a translation from **Show preview for translation**, or enter a Contact in **Show preview for contact** to see the Email as that Contact would receive it. When you select a Contact, Mautic fills in the :ref:`tokens <Email tokens>` in the content — placeholders that pull in that Contact's field values, such as their first name — so you can check the personalized version. Mautic reloads the preview automatically to match the version you've selected.

Reviewer: Please investigate the code change. Does it make any changes for token? If not, we can drop the token part. Because I feel like this info about how it works is not necessary as it's too detail. This whole section is about previewing email content and we can keep it simple as long as it covers the code changes.

  :width: 400
  :alt: Screenshot showing A/B and Translation preview

> The rendered preview on the Email details page reloads automatically when you switch between variants and translations. For more information, see :ref:`email content preview`.

Reviewer: ```suggestion
The rendered preview on the Email details page reloads automatically when you switch between variants and translations. For more information, see :ref:`Email content preview`.
```
